src/analysisd/alerts/exec.c

```diff
@@ -106,46 +107,145 @@ void OS_Exec(int execq, int *arq, const Eventinfo *lf, const active_response *ar
 
     /* Active Response to the forwarder */
     else if ((Config.ar & REMOTE_AR)) {
-        int rc;
-        /* If lf->location start with a ( was generated by remote agent and its
-         * ID is included in lf->location if missing then it must have been
-         * generated by the local analysisd, so prepend a false id tag */
-        if (lf->location[0] == '(') {
-            snprintf(exec_msg, OS_SIZE_1024,
-                     "%s %c%c%c %s %s %s %s %ld.%ld %d %s %s %s",
-                     lf->location,
-                     (ar->location & ALL_AGENTS) ? ALL_AGENTS_C : NONE_C,
-                     (ar->location & REMOTE_AGENT) ? REMOTE_AGENT_C : NONE_C,
-                     (ar->location & SPECIFIC_AGENT) ? SPECIFIC_AGENT_C : NONE_C,
-                     ar->agent_id != NULL ? ar->agent_id : "(null)",
-                     ar->name,
-                     user,
-                     ip,
-                     (long int)lf->time.tv_sec,
-                     __crt_ftell,
-                     lf->generated_rule->sigid,
-                     lf->location,
-                     filename ? filename : "-",
-                     extra_args ? extra_args : "-");
+
+        /* We need to identify the version of Wazuh installed on the agents because 
+         * if the agents version is >4.2, it is necessary to send legacy 
+         * message in string format. Instead if the agent version is <=4.2, 
+         * we need to send the new message in string format.*/
+
+        if (ar->location & ALL_AGENTS) {
+
+            int sock = -1;
+            int *id_array =  NULL;
+            
+            id_array = wdb_get_all_agents(FALSE, &sock);
+            if(!id_array){
+                merror("\n Unable to get agent info");
+                return;
+            }
+
+            for (size_t i = 0; id_array[i] != -1; i++) {
+                cJSON *json_agt_info = NULL;
+                cJSON *json_agt_version = NULL;
+                char *agt_version = NULL;
+
+                json_agt_info = wdb_get_agent_info(id_array[i], &sock);
+                if (!json_agt_info) {
+                    merror("\n Unable to get agent info");
```

Comment:
Replace this log with:
`merror("Failed to get agent '%d' information from Wazuh DB.", id_array[i]);`
Also, replace the `return` with a `continue` in the next line to avoid stop processing the AR.

Reply:
Change accepted.